openstack/identity/v3/policies/requests.go

```diff
@@ -39,3 +39,52 @@ func List(client *gophercloud.ServiceClient, opts ListOptsBuilder) pagination.Pa
 		return PolicyPage{pagination.LinkedPageBase{PageResult: r}}
 	})
 }
+
+// CreateOptsBuilder allows extensions to add additional parameters to
+// the Create request.
+type CreateOptsBuilder interface {
+	ToPolicyCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts provides options used to create a policy.
+type CreateOpts struct {
+	// Type is the MIME media type of the serialized policy blob.
+	Type string `json:"type" required:"true"`
+
+	// Blob is the policy rule as a serialized blob.
+	Blob string `json:"blob" required:"true"`
```

Comment:
Orchestration and Container also have support for fields which take JSON or YAML structures and we use `[]byte` as the type. For consistency, let's change this to `[]byte`.

https://github.com/gophercloud/gophercloud/blob/master/openstack/container/v1/capsules/requests.go#L40-L52

I don't see a huge need to have a separate `Template` struct/field/type here. Mainly because I'm not sure all of the MIME types that the policy API supports and so adding validation and such might be difficult. But it's up to you if you want to go that route.

Reply:
The Keystone service does not check the validation of **blob** and **type**. Arbitrary string is accepted. The only limitation is the length of **type**.

https://github.com/openstack/keystone/blob/master/keystone/policy/schema.py#L14

So I think there is no need to add a **Template** type. Instead, we can check the length of **type**.

I agree to change type of **Blob** from string to []byte for consistency.